src/core/recv_buffer.c

```diff
@@ -907,7 +907,12 @@ QuicRecvBufferDrain(
     }
 
     do {
-        if ((uint64_t)RecvBuffer->ReadLength > DrainLength) {
+        if ((uint64_t)RecvBuffer->ReadLength > DrainLength ||
```

Comment:
This isn't quite the right fix. It should still call `QuicRecvBufferFullDrain` to indicate it drained everything that was read, but `QuicRecvBufferFullDrain` should be handling the fact that just because it read everything doesn't mean there isn't more after a gap to read.

Reply:
I am looking into what the right fix should be.

Reply:
We talked over IM and maybe using the partial drain is the right way to go. We'd just kind of redefine 'partial drain' here to indicate it's not partial of the read length, but of all (possibly discontiguous) data in the buffer.